core/lib/basic_types/src/basic_fri_types.rs

```diff
@@ -31,10 +31,12 @@ impl Eip4844Blobs {
     pub fn blobs(self) -> Vec<Blob> {
         self.blobs
     }
-}
 
-impl From<Vec<u8>> for Eip4844Blobs {
-    fn from(blobs: Vec<u8>) -> Self {
+    pub fn encode(self) -> Vec<u8> {
+        self.blobs.iter().flatten().copied().collect()
```

Comment:
```suggestion
        let blobs_len = self.blobs.len() * EIP_4844_BLOB_SIZE;
        self.blobs
            .into_iter()
            .fold(Vec::with_capacity(blobs_len), |mut acc, v| {
                acc.extend(v);
                acc
            })
```

can save a bit of latency here by not reallocating.